ert3/engine/_run.py

```diff
@@ -9,15 +9,21 @@ def _prepare_experiment(
     experiment_name: str,
     ensemble: ert3.config.EnsembleConfig,
     ensemble_size: int,
+    parameters_config: ert3.config.ParametersConfig,
 ) -> None:
     if ert3.workspace.experiment_has_run(workspace_root, experiment_name):
         raise ValueError(f"Experiment {experiment_name} have been carried out.")
 
-    parameter_names = [elem.record for elem in ensemble.input]
+    parameter_names = {elem.record for elem in ensemble.input}
+    parameters = {
+        param.name: list(param.variables)
+        for param in parameters_config
+        if param.name in parameter_names
+    }
     ert3.storage.init_experiment(
         workspace=workspace_root,
         experiment_name=experiment_name,
-        parameters=parameter_names,
+        parameters=parameters,
```

Comment:
Should we call it `parameter_layout` or similar?

Reply:
Layout isn't descriptive (ie, the word doesn't mean anything by itself, except in GUIs). `parameter_records`?

Reply:
`parameter_variables`?

Reply:
According to `ert3` each element of the structure is providing the name of a record, together with the index of that particular record. That is the currently used notation within `ert3` and I think we should stick with that until we agree on something else ™️

`record_index` makes sense, `parameter_index` can make sense. `parameter` or `record` followed by structure, layout, blueprint, schema or a similar term might also make sense...